### ISSUE TITLE
App blank after deploying

### ISSUE THREAD

**Issue by datawookie** [dominodatalab/low-code-assistant#22]
I seem to have successfully deployed an app.

![image](https://user-images.githubusercontent.com/6134409/196974571-9c1ebf5a-4887-4b13-8a1a-3cc72bff2077.png)

The layout is slightly different from what I expected based on the existing documentation. However, I guess that the deployment was successful because the **View App** button is no longer greyed out.

However, when I press the **View App** button I was taken to a blank page.

![image](https://user-images.githubusercontent.com/6134409/196975676-aab19379-bc4a-42fb-9a69-25073aadb177.png)

This didn't seem right, so I went back to the deploy page and pressed the **Deploy App** button again (it was also no longer greyed out). The same process was repeated. This time when I pressed the **View App** button I was taken to a page which indicated "There Is No App Associated with This Project".

![image](https://user-images.githubusercontent.com/6134409/196976029-bd5ef63b-4a9d-49c2-9369-52b92f90d4ad.png)

**Comment by mariobuikhuizen** [comment]
The second image is because it sometimes takes a bit longer for the app to be really available. Trying to refresh the page a bit later should help.

I will investigate the blank page of the app.

**Comment by datawookie** [comment]
I left that tab open and this is what I got.

![image](https://user-images.githubusercontent.com/6134409/196980584-a7997c76-a938-4543-b456-31f8b51c9f53.png)

Here's the complete traceback.

```
Traceback (most recent call last):
  File "/opt/conda/lib/python3.9/urllib/request.py", line 1346, in do_open
    h.request(req.get_method(), req.selector, req.data, headers,
  File "/opt/conda/lib/python3.9/http/client.py", line 1285, in request
    self._send_request(method, url, body, headers, encode_chunked)
  File "/opt/conda/lib/python3.9/http/client.py", line 1331, in _send_request
    self.endheaders(body, encode_chunked=encode_chunked)
  File "/opt/conda/lib/python3.9/http/client.py", line 1280, in endheaders
    self._send_output(message_body, encode_chunked=encode_chunked)
  File "/opt/conda/lib/python3.9/http/client.py", line 1040, in _send_output
    self.send(msg)
  File "/opt/conda/lib/python3.9/http/client.py", line 980, in send
    self.connect()
  File "/opt/conda/lib/python3.9/http/client.py", line 1447, in connect
    super().connect()
  File "/opt/conda/lib/python3.9/http/client.py", line 946, in connect
    self.sock = self._create_connection(
  File "/opt/conda/lib/python3.9/socket.py", line 844, in create_connection
    raise err
  File "/opt/conda/lib/python3.9/socket.py", line 832, in create_connection
    sock.connect(sa)
TimeoutError: [Errno 110] Connection timed out

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/opt/conda/lib/python3.9/site-packages/low_code_assistant/nb_app.py", line 45, in exec_nb
    exec(ast, user_ns)
  File ".lca_deployed.ipynb input cell 2", line 3, in <module>
  File "/opt/conda/lib/python3.9/site-packages/pandas/util/_decorators.py", line 311, in wrapper
    return func(*args, **kwargs)
  File "/opt/conda/lib/python3.9/site-packages/pandas/io/parsers/readers.py", line 680, in read_csv
    return _read(filepath_or_buffer, kwds)
  File "/opt/conda/lib/python3.9/site-packages/pandas/io/parsers/readers.py", line 575, in _read
    parser = TextFileReader(filepath_or_buffer, **kwds)
  File "/opt/conda/lib/python3.9/site-packages/pandas/io/parsers/readers.py", line 933, in __init__
    self._engine = self._make_engine(f, self.engine)
  File "/opt/conda/lib/python3.9/site-packages/pandas/io/parsers/readers.py", line 1217, in _make_engine
    self.handles = get_handle(  # type: ignore[call-overload]
  File "/opt/conda/lib/python3.9/site-packages/pandas/io/common.py", line 670, in get_handle
    ioargs = _get_filepath_or_buffer(
  File "/opt/conda/lib/python3.9/site-packages/pandas/io/common.py", line 339, in _get_filepath_or_buffer
    with urlopen(req_info) as req:
  File "/opt/conda/lib/python3.9/site-packages/pandas/io/common.py", line 239, in urlopen
    return urllib.request.urlopen(*args, **kwargs)
  File "/opt/conda/lib/python3.9/urllib/request.py", line 214, in urlopen
    return opener.open(url, data, timeout)
  File "/opt/conda/lib/python3.9/urllib/request.py", line 517, in open
    response = self._open(req, data)
  File "/opt/conda/lib/python3.9/urllib/request.py", line 534, in _open
    result = self._call_chain(self.handle_open, protocol, protocol +
  File "/opt/conda/lib/python3.9/urllib/request.py", line 494, in _call_chain
    result = func(*args)
  File "/opt/conda/lib/python3.9/urllib/request.py", line 1389, in https_open
    return self.do_open(http.client.HTTPSConnection, req,
  File "/opt/conda/lib/python3.9/urllib/request.py", line 1349, in do_open
    raise URLError(err)
urllib.error.URLError: <urlopen error [Errno 110] Connection timed out>
```

**Comment by datawookie** [comment]
@mariobuikhuizen it's just occurred to me that the setup for the plot in the app might be relevant. I'm pulling data from a GitHub Gist into a Pandas Data Frame. Then using Plotly Express to generate a scatter plot. When I tried to do this initially I ran into an issue because the instance type did not have egress access. However, not I'm running on an instance with egress. That traceback looks very much like it might be related to downloading the data...

**Comment by mariobuikhuizen** [comment]
Ah, right. The app uses the default hardware-tier which might not have egress.

**Comment by datawookie** [comment]
That's a bit of a limitation for people developing "useful" apps which might access external resources.

How about being able to choose the hardware tier when you launch the app and defaulting to the lowest tier?

**Comment by mariobuikhuizen** [comment]
I'm looking into being able to set the tier, or at least automatically use the same tier as the notebook.

**Comment by mariobuikhuizen** [comment]
This is fixed with: bdb490b6

**Comment by datawookie** [comment]
Thanks, @mariobuikhuizen! This feels like a pretty important feature! :+1:
